### PR TITLE
Transform into library

### PR DESCRIPTION
**Pull request by robin-nitrokey**
- Move logic from binary into library
- Transform binary into example
- Introduce feature flag for `ctaphid` support

### CODE REVIEW

File: Cargo.toml
Comment (szszszsz) [inline Cargo.toml:21]: Question: is having `pretty_env_logger` in `dev-dependencies` will not block me from getting logs from this lib while configuring in the dependents?

File: Cargo.toml
Comment (szszszsz) [inline Cargo.toml]: Minor: let's move this branch to Nitrokey/trussed space

**Review by szszszsz — APPROVED** [review]
Thank you for cleaning this up!

File: Cargo.toml
Comment (szszszsz) [reply Cargo.toml]: Since this is a side branch anyway, perhaps even better would be to keep it trussed-dev/trussed space instead. Let's discuss that on the next occassion.

File: Cargo.toml
Comment (robin-nitrokey) [reply Cargo.toml:21]: No, libraries should only depend on `log` and use its logging macros (as we do).  Binaries can then decide how the log messages should be handled by selecting a logger implementation, e. g. `pretty_env_logger`.

File: Cargo.toml
Comment (robin-nitrokey) [reply Cargo.toml]: I would just wait until it is merged as this PR is not really urgent.

**Pull request merged**
